discovery_engine/lib/src/domain/document_manager.dart

```diff
@@ -82,6 +86,24 @@ class DocumentManager {
       throw StateError('id $id does not have active data attached');
     }
 
+    // update reacted sources repo if necessary
+    if (userReaction != UserReaction.neutral) {
+      final source = doc.resource.sourceDomain;
+      final like = userReaction == UserReaction.positive;
+      final sourcesOpp = await _sourceRepo.fetchByReaction(!like);
```

Comment:
❔ would it make sense to have a method like `fetchBySource` to avoid iterating over the sources?
```dart
    if (userReaction != UserReaction.neutral) {
      final docSource = doc.resource.sourceDomain;
      final like = userReaction == UserReaction.positive;
      SourceReacted? source = await _sourceRepo.fetchBySource(docSource.value);
      if (source != null && source.liked != like) {
        await _sourceRepo.remove(docSource);
        source = null;
      }
      if (source == null) {
        await _sourceRepo.save(SourceReacted(docSource, like));
      } else {
        await _sourceRepo.save(source..update());
      }
    }
```

Reply:
well spotted, yes it looks like a `fetchBySource` should be sufficient here!